unitstatushandler/softwaremanager.go

```diff
@@ -1341,7 +1377,29 @@ func (manager *softwareManager) runInstances(newServices []string) (runErr error
 	return nil
 }
 
-func (manager *softwareManager) updateUnitConfig(ctx context.Context) (unitConfigErr error) {
+func (manager *softwareManager) updateNodes() (nodesErr error) {
+	log.Debug("Update nodes status")
+
+	for _, nodeStatus := range manager.CurrentUpdate.NodesStatus {
+		if nodeStatus.Status == cloudprotocol.NodeStatusPaused {
+			if err := manager.nodeManager.PauseNode(nodeStatus.NodeID); err != nil && nodesErr == nil {
+				log.WithField("nodeID", nodeStatus.NodeID).Errorf("Can't pause node: %v", err)
+
+				nodesErr = aoserrors.Wrap(err)
+			}
+		} else {
```

Comment:
It's not clear why we should send ResumeNode on every status update where status != cloudprotocol.NodeStatusPaused